src/parser/mod.rs

```diff
@@ -5200,12 +5200,26 @@ impl<'a> Parser<'a> {
         // parse: [ argname ] argtype
         let mut name = None;
         let mut data_type = self.parse_data_type()?;
-        if let DataType::Custom(n, _) = &data_type {
-            // the first token is actually a name
-            match n.0[0].clone() {
-                ObjectNamePart::Identifier(ident) => name = Some(ident),
+
+        // It may appear that the first token can be converted into a known
+        // type, but this could also be a collision as some types are only
+        // present in some dialects and therefore some type reserved keywords
+        // may be freely used as argument names in other dialects.
+
```

Comment:
```suggestion
```
I think we could remove this part to keep the comment smaller/focused. This part seems to be describing the current bug, which wouldn't be obvious when reading going forward since the bug would have been fixed. (smaller text also makes it more likely to be read in general)